pkg/reloader/reloader.go

```diff
@@ -280,7 +283,11 @@ func (r *Reloader) Watch(ctx context.Context) error {
 		err := r.apply(initialSyncCtx)
 		initialSyncCancel()
 		if err != nil {
-			return err
+			if !errors.Is(err, expandEnvError) {
+				return err
+			}
+			r.configApplyErrors.Inc()
+			level.Error(r.logger).Log("msg", "expand environment variables", "err", err)
```

Comment:
So by logging out, sidecar doesn't fail, but there would be errors on Prometheus?

Reply:
Ah, due to unset variables? Referencing the parent issue, there's a need to allow the reloader to not crash on such instances, as is the case for all `>1` file modifications (and reloads), and thus align environment expansion errors to that [same behavior](https://github.com/thanos-io/thanos/pull/7429/files#diff-18817f312c728d47e162f19e1ea8cb65aec318bc1b330a4a77efcd18d182a77dR350-R353).